expr/expr.go

```diff
@@ -801,3 +823,93 @@ func compileArithmetic(lhsFunc, rhsFunc NativeEvaluator, operator string) (Nativ
 		}
 	}, nil
 }
+
+func getNthFromContainer(container zcode.Bytes, idx uint) (zcode.Bytes, error) {
+	iter := zcode.Iter(container)
+	var i uint = 0
+	for ; !iter.Done(); i++ {
+		zv, _, err := iter.Next()
+		if err != nil {
+			return nil, err
+		}
+		if i == idx {
+			return zv, nil
+		}
+	}
+	return nil, ErrIndexOutOfBounds
+}
+
+func compileArrayIndex(lhsFunc, rhsFunc NativeEvaluator, operator string) (NativeEvaluator, error) {
+	return func(rec *zng.Record) (NativeValue, error) {
+		lhs, err := lhsFunc(rec)
+		if err != nil {
+			return NativeValue{}, err
+		}
+
+		var aType *zng.TypeArray
+		var ok bool
+		if aType, ok = lhs.typ.(*zng.TypeArray); !ok {
+			return NativeValue{}, ErrIncompatibleTypes
+		}
+
+		rhs, err := rhsFunc(rec)
+		if err != nil {
+			return NativeValue{}, err
+		}
+
+		var idx uint
+		switch rhs.typ.ID() {
+		case zng.IdByte, zng.IdUint16, zng.IdUint32, zng.IdUint64:
+			idx = uint(rhs.value.(uint64))
+		case zng.IdInt16, zng.IdInt32, zng.IdInt64:
```

Comment:
Any reason not to do `switch rhs.type {`?

Reply:
Not sure what you're suggesting, `rhs.type` isn't a thing.  You mean `rhs.typ.(type)` ?  The immediate problem with that is that it doesn't resolve aliases (there was a lengthy discussion about this on a different PR)

Reply:
I mean `rhs.typ`. And I'd forgotten about aliases. 🤦‍♂